### DISCUSSION TITLE
Decryption rules for google ads traffic

### DISCUSSION THREAD

**Discussion by geeky01adarsh** [emanuele-f/PCAPdroid#472] [Q&A] [answered]
I tried capturing the data along with decryption mode using ssl keys generated through MiTM way. 
What I found was while most of the traffic were decrypted, still all data from google ads traffic were encrypted.
Is there any way I can check for the validation of these data as well?

**Comment by emanuele-f** [comment]
Browser traffic or other apps? For browsers, root is not necessary while for other apps usually it is. See https://emanuele-f.github.io/PCAPdroid/tls_decryption#34-caveats-and-possible-solutions for more details. Apps from big companies will require specific methods to bypass protections. On the Internet you may find some useful info

**Reply by geeky01adarsh** [reply]
Hi @emanuele-f ,
I was trying to look through IMO SNS traffic and understand what data it shares with advertising agencies.
Is there any way I can export the keys to analyze them over WireShark or something, to have a better readability insight?

**Reply by emanuele-f** [reply, marked as answer]
The most convenient way is to use the [PCAPng dump format](https://emanuele-f.github.io/PCAPdroid/paid_features#53-pcapng-format), which includes both traffic and decryption keys in a single file. Otherwise you will have to export the SSLKEYLOG and load both it and the PCAP file in Wireshark.
